packages/suite/src/views/wallet/transactions/TransactionList/WalletTransactionList.tsx

```diff
@@ -0,0 +1,66 @@
+import { useState } from 'react';
+
+import styled from 'styled-components';
+
+import { Card, Column } from '@trezor/components';
+import { typography } from '@trezor/theme';
+
+import { Translation } from 'src/components/suite';
+import { Account, WalletAccountTransaction } from 'src/types/wallet';
+
+import { TransactionList } from './TransactionList';
+import { useVisibleTransactionsRetriever } from './useFetchTransactions';
+
+const NoResults = styled.div`
+    text-align: center;
+    color: ${({ theme }) => theme.legacy.TYPE_DARK_GREY};
+    ${typography.hint}
+`;
+
+export const NoVisibleTransactions = () => (
+    <Card>
+        <NoResults>
+            <Column alignItems="center">
+                <Translation id="TR_NO_VISIBLE_TRANSACTIONS" />
+            </Column>
+        </NoResults>
+    </Card>
+);
+
+interface TransactionListProps {
+    symbol: WalletAccountTransaction['symbol'];
+    account: Account;
+    customTotalItems?: number;
+    isExportable?: boolean;
+}
+
+export const WalletTransactionList = ({
+    account,
+    symbol,
+    customTotalItems,
+    isExportable = true,
+}: TransactionListProps) => {
+    // NOTE: The number of the displayed pages may be different from the number of the pages for all transactions
+    const [visiblePages, setVisiblePages] = useState(1);
```

Comment:
This tracks the paging according to the visible transactions, it is not the same as if you derived paging from the total.